presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/Lookup.java

```diff
@@ -35,15 +34,17 @@
     @Deprecated
     default PlanNode resolve(PlanNode node)
     {
-        return getOnlyElement(resolveGroup(node));
+        if (node instanceof GroupReference) {
```

Comment:
What's the rationale for having different semantics for `resolve` and `resolveGroup`? The arguments in commit message also apply for `Lookup::resolve` (while it's depricated it is still valid method of the interface).

The previous semantics of those two methods seems is a bit more natural:
```
* resolveGroup: resolve a group (whatever it means)
* resolve: the same as resolveGroup, but assume that each group is singleton (valid only for iterative)
```

In fact, current semantics of `resolve` is rather convenience over strict semantics. Do we want to get rid of this convenience eventually? E.g: we can test algorithms on raw trees without any group references.

Reply:
Agree, but I do not want to change since it is deprecated and it is widely used (various visitors).

Regarding tests, and convenience. IMO it should be know what is what in input, rule is design to work with group references so there should group references in tests as well (to be as close to production code as possible).  Hence, method should remain as at same convenience level as it is today, but now you need to be careful (think) when using it.